src/components/App.js

```diff
@@ -1,21 +1,75 @@
 /* eslint-disable no-unused-vars */
 /* eslint-disable import/extensions */
+/* eslint-disable no-restricted-globals */
 
-import React from 'react';
+import React, { Component } from 'react';
 import './App.css';
 
 import ButtonPanel from './ButtonPanel';
 import Display from './Display';
 
 import calculate from '../logic/calculate.js';
 
-function App() {
-  return (
-    <div className="App">
-      <Display />
-      <ButtonPanel />
-    </div>
-  );
+class App extends Component {
+  constructor() {
+    super();
+    this.state = {
+      total: '',
+      next: '',
+      operation: null,
+      result: '',
+    };
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick(evt) {
+    const buttonName = evt.target.textContent;
+    const {
+      total, next, operation, result,
+    } = this.state;
+    if (operation === null && !isNaN(buttonName) && next === '') {
+      this.setState({
+        total: `${total + buttonName}`,
+        result: `${result + buttonName}`,
+      });
+    } else if (total !== '' && operation === null && buttonName !== '.') {
+      this.setState({
+        operation: buttonName,
+        result: `${`${result} ${buttonName} `}`,
+      });
+    } else if (!isNaN(buttonName)) {
+      this.setState({
+        next: `${next + buttonName}`,
+        result: `${result + buttonName}`,
+      });
+    }
```

Comment:
[Optional]

It would be better to keep the calculator logic away from the App component since we have a logic folder that should contain all the necessary logic. The App component only needs to call the `calculate` function and the `calculate` should handle all logic.

I marked this as optional since @shloch didn't mention it in the first review, I will recommend you refactor the code to move the logic to the `calculate` function.

Reply:
Done